### ISSUE TITLE
How to compile the GPU

### ISSUE THREAD

**Issue by huxin-cc** [JiayuanWang-JW/YOLOv8-multi-task#24]
Hi Jiayuan,
When I want to train the model, I set the `device = [0,1,2,3]` in the train.py, for there are 4 available in the server. 
![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/65339149/1c2877d5-ca0d-4de7-a4b2-0fa24a3c4122)
However, it appeared the problem of not seeing CUDA devices by torch. 
![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/65339149/3a7c3aee-b249-401d-854b-81931edfb2b4)
Could you please help to solve this problem?

**Comment by huxin-cc** [comment]
Hi Jiayuan, I also found that when I try to test the model, and set the `show_labels=True` in the predict.py , the pretrained model will label the car to the person. Does it mean that I should custom the model myself to give it the ability to detect the cars?
![image](https://github.com/JiayuanWang-JW/YOLOv8-multi-task/assets/65339149/97fde7d6-a55e-4c3d-8944-6eec75ea53d8)

**Comment by huxin-cc** [comment]
> Hi Jiayuan, When I want to train the model, I set the `device = [0,1,2,3]` in the train.py, for there are 4 available in the server. ![image](https://private-user-images.githubusercontent.com/65339149/288296546-1c2877d5-ca0d-4de7-a4b2-0fa24a3c4122.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NzEwOTgsIm5iZiI6MTcwMjQ3MDc5OCwicGF0aCI6Ii82NTMzOTE0OS8yODgyOTY1NDYtMWMyODc3ZDUtY2EwZC00ZGU3LWE0YjItMGZhMjRhM2M0MTIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDEyMzMxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMWNkNjk0MWQ5Y2E2ZTQyZWNmNTllNGQyMjE2OTlkYjg0Njk0OWFjYmFmNTI0MDc0NDcwNTEwMDI5YjRiNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.peTeuFsBxOpjN2D6vig8Vo8nvxE2-Q_cLGCi1mOdLB4) However, it appeared the problem of not seeing CUDA devices by torch. ![image](https://private-user-images.githubusercontent.com/65339149/288295380-3a7c3aee-b249-401d-854b-81931edfb2b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NzEwOTgsIm5iZiI6MTcwMjQ3MDc5OCwicGF0aCI6Ii82NTMzOTE0OS8yODgyOTUzODAtM2E3YzNhZWUtYjI0OS00MDFkLTg1NGItODE5MzFlZGZiMmI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDEyMzMxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MjMwNTYxMmIzZWQzZjdiNDJhYmM3OWMyNTQ3Njk0ZjRmNjYwM2U0ZDEyZmM5M2FmZjYxYTI0MmE2NmJhMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uviuHogtBy41PTuC0CwrvMM3JRkHqQ7BIsPvwdXVw44) Could you please help to solve this problem?

I solved this problem by running in the docker. Also I reinstall the cuda toolkit, the pytorch, torchvision, and torchaudio in the docker. Then I can train succeessfully.

**Comment by huxin-cc** [comment]
> Hi Jiayuan, I also found that when I try to test the model, and set the `show_labels=True` in the predict.py , the pretrained model will label the car to the person. Does it mean that I should custom the model myself to give it the ability to detect the cars? ![image](https://private-user-images.githubusercontent.com/65339149/288325789-97fde7d6-a55e-4c3d-8944-6eec75ea53d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwMDg4MDksIm5iZiI6MTcwNDAwODUwOSwicGF0aCI6Ii82NTMzOTE0OS8yODgzMjU3ODktOTdmZGU3ZDYtYTU1ZS00YzNkLTg5NDQtNmVlYzc1ZWE1M2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMxVDA3NDE0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjZjU5NGNmMjllMDM0ODJkMzVhZWZhOWMyMDcyMTk1MGIzMzRmMGJhNjYzYzVhMWVkOWY0ZjQyY2JhMGFmZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GrLB6e1MUA3bXECn7iwvV3VlpTJEpHU3oRibl6_BExQ)

I solved this problem by modifying the `nc` to 10 in the `yolov8-bdd-v4-one-dropout-individual-n.yaml`.

**Comment by huxin-cc** [comment]
I solved all problems by modifying the configuration files. There are no problems in author's code.
